lib/mistral_ai_summary_example/settings_dialogs.dart

```diff
@@ -0,0 +1,262 @@
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 500',
+            ),
+          ),
+          const SizedBox(height: 16),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Unlimited',
+              ),
+              Switch(
+                value: _maxTokensController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class RandomSeedDialog extends StatefulWidget {
+  const RandomSeedDialog({
+    super.key,
+  });
+
+  @override
+  State<RandomSeedDialog> createState() => _RandomSeedDialogState();
+}
+
+class _RandomSeedDialogState extends State<RandomSeedDialog> {
+  final TextEditingController _randomSeedController = TextEditingController();
+
+  @override
+  void initState() {
+    _randomSeedController.text =
+        context.read<SummarySettingsModel>().settings.randomSeed?.toString() ??
+            '';
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _randomSeedController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return CommonSettingsDialog(
+      title: 'Random Seed',
+      description: 'Write a number if you want to set a random seed',
+      onSettingChanged: () {
+        final randomSeed = _randomSeedController.text.isEmpty
+            ? null
+            : int.tryParse(_randomSeedController.text);
+        context.read<SummarySettingsModel>().setRandomSeed(randomSeed);
+        Navigator.of(context).pop();
+      },
+      child: Column(
+        children: [
+          const SizedBox(height: 24),
+          TextField(
+            controller: _randomSeedController,
+            onChanged: (_) => setState(() {}),
+            decoration: InputDecoration(
+              border: const OutlineInputBorder(),
+              suffixIcon: InkWell(
+                borderRadius: BorderRadius.circular(20),
+                onTap: () => setState(_randomSeedController.clear),
+                child: const Icon(Symbols.cancel),
+              ),
+              hintText: 'e.g. 123',
+            ),
+          ),
+          const SizedBox(height: 16),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              const Text(
+                'Unset',
+              ),
+              Switch(
+                value: _randomSeedController.text.isEmpty,
+                onChanged: null,
+                activeTrackColor: Theme.of(context).colorScheme.primary,
+              ),
+            ],
+          ),
+          const SizedBox(height: 24),
+        ],
+      ),
+    );
+  }
+}
+
+class CommonSettingsDialog extends StatelessWidget {
```

Comment:
Not sure if `AlertDialog` would give you the same structure as this widget.

Also, the name could be more like `BaseSettingsDialog` with some description that this is the base of any dialog to update settings.

Reply:
`AlertDialog` would probably give me a similar or the same structure, but I am unsure if we should use it in this context. As [docs](https://api.flutter.dev/flutter/material/AlertDialog-class.html) says alert dialog should be used for some critical information or simple input like yes/no.

I agree that name can be better, renamed in [3a52988](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/3a52988d17c109e3c5eab52904d2a72205c7ec68)

Reply:
Docs say `An alert dialog (also known as a basic dialog) `.  So seems appropriate for me :D

Reply:
Refactored in [73eb97c](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/73eb97c09a3f1147e792f9388ef7b2867cedf592)

I also explicitly set the alert dialog and list tile color in `ModelSettingsDialog` (renamed in [5a71018](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/5a71018ec9001492d0b1133cf191693b9ef70452) ) because after using `AlertDialog`, the list tiles were white

Reply:
Hmm let's try to find a way to change those colors in theme instead directly on the widgets.

Reply:
Resolved in [64a1cb3](https://github.com/nomtek/flutter_ai_examples/pull/30/commits/64a1cb350f974071c5213a0b636aeb86ebb60879)